backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/handlingunits/rest_api/HandlingUnitsService.java

```diff
@@ -519,4 +570,91 @@ private HuId resolveHUIdentifier(@NonNull final JsonSetClearanceStatusRequest.Js
 				.appendParametersToMessage()
 				.setParameter("huIdentifier", jsonHuIdentifier);
 	}
+
+	@NonNull
+	private Set<HuId> resolveHuIds(@NonNull final JsonGetByQRCodeRequest request)
+	{
+		final HUQRCodeAssignment huqrCodeAssignment = huQRCodeService
+				.getHUAssignmentByQRCode(HUQRCode.fromGlobalQRCodeJsonString(request.getQrCode()))
```

Comment:
minor: imho getHUAssignmentByQRCode shall not accept null as param...

Reply:
it doesn't, but the `getHUAssignmentByQRCode` returns an Optional